packages/frontend/amp/components/Header.tsx

```diff
@@ -174,10 +175,11 @@ const pillarLinks = (pillars: PillarType[], activePillar: Pillar) => (
 const supportLink =
     'https://support.theguardian.com/?INTCMP=header_support&acquisitionData=%7B%22source%22:%22GUARDIAN_WEB%22,%22componentType%22:%22ACQUISITIONS_HEADER%22,%22componentId%22:%22header_support%22%7D';
 
-const Header: React.SFC<{
+const Header: React.FunctionComponent<{
```

Comment:
This is why I don't like hooks :( (because we can no longer know that function components are stateless).

I don't know if there is anyway to guarantee they are stateless now by the type signature?

Reply:
I tend to agree, what used to signify an idempotent props -> render output function, could now entail something quite the opposite. I think the problem comes from the fact that Hook functions rely on mutable state.

Also, reminds about [this Tweet](https://twitter.com/dan_abramov/status/1057625147216220162?lang=en).

Reply:
Ah interesting tweet! Bit surprising that he is arguing we should never have assumed functional components were stateless given that there has been a clear class/function divide for so long.